Lib/test/datetimetester.py

```diff
@@ -2425,6 +2425,22 @@ def test_strptime(self):
                 self.assertEqual(expected, got)
 
         strptime = self.theclass.strptime
+
```

Comment:
There's actually a https://github.com/python/cpython/blob/master/Lib/test/test_strptime.py file for tests relating to `strptime`.

Reply:
I am not a *huge* fan of the `test_strptime` thing, which tests `_strptime` directly rather than testing the public API function(s) `datetime.strptime` and `time.strptime`.

The reasoning for this is that the tests in `datetimetester.py` run against the C and `datetime` implementations per [PEP 399](https://www.python.org/dev/peps/pep-0399/), and they are parametrized to be run against subclasses of `datetime`.

In this case, I think it's unlikely that these tests will be affected by either of these things, but I also don't think the cost is particularly high, so I lean in favor of testing in `datetimetester.py` rather than `test_strptime.py` unless there's a compelling reason to do otherwise.